src/test/java/org/ow2/proactive/catalogclient/service/CatalogObjectServiceTest.java

```diff
@@ -0,0 +1,163 @@
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.catalogclient.service;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.charset.Charset;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.ow2.proactive.catalogclient.model.CatalogObject;
+
+import com.google.common.io.Files;
+
+
+@RunWith(MockitoJUnitRunner.class)
+public class CatalogObjectServiceTest {
+
+    @Mock
+    private RemoteObjectService remoteObjectService;
+
+    @Mock
+    private ServiceConfiguration serviceConfiguration;
+
+    @InjectMocks
+    private CatalogObjectService catalogObjectService;
+
+    private static final String CATALOG_URL = "http://localhost:8080/catalog";
+
+    private static final String REMOTE_URL = "localhost:8080/test/example.txt";
+
+    private static final String REMOTE_URL2 = "localhost:8080/test/example2.txt";
+
+    private static final String REMOTE_VALUE = "remoteValue";
+
+    private static final String REMOTE_VALUE2 = "remoteValue2";
+
+    private static final String CATALOG_RESOURCE_URL = CATALOG_URL + "/buckets/3/resources/bobot";
+
+    private static final String CATALOG_RESOURCE_URL_RAW = CATALOG_RESOURCE_URL + "/raw";
+
+    @Test
+    public void getMetadataURLWithEndingSlash() {
+        assertThat(catalogObjectService.getURL(CATALOG_URL + "/", 3, "bobot", false)).isEqualTo(CATALOG_RESOURCE_URL);
+    }
+
+    @Test
+    public void getMetadataURLWithoutEndingSlash() {
+        assertThat(catalogObjectService.getURL(CATALOG_URL, 3, "bobot", false)).isEqualTo(CATALOG_RESOURCE_URL);
+
+    }
+
+    @Test
+    public void getRawURLWithEndingSlash() {
+        assertThat(catalogObjectService.getURL(CATALOG_URL + "/",
+                                               3,
+                                               "bobot",
+                                               true)).isEqualTo(CATALOG_RESOURCE_URL_RAW);
+
+    }
+
+    @Test
+    public void getRawURLWithoutEndingSlash() {
+        assertThat(catalogObjectService.getURL(CATALOG_URL, 3, "bobot", true)).isEqualTo(CATALOG_RESOURCE_URL_RAW);
+    }
+
+    @Test
+    public void getRawCatalogObjectTest() {
+        String returnedObject = "{ \"raw\":\"value\"}";
+        String objectURL = CATALOG_RESOURCE_URL_RAW;
+        when(remoteObjectService.sendRequest(objectURL, String.class)).thenReturn(returnedObject);
+        String result = catalogObjectService.getRawCatalogObject(CATALOG_URL, 3, "bobot");
+        assertThat(result).isEqualTo(returnedObject);
+        verify(remoteObjectService, times(1)).sendRequest(objectURL, String.class);
+    }
+
+    @Test
+    public void getCatalogObject() {
```

Comment:
can you be more specific. Like testThatCatalogObjectIsCorrectlyReceived...... or something like that